---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---
# This is 16S data from 2023 adn 2024 Chicago River Pilot and wrangle for APAL samples for heat tolerance study. 

Data was first run with no sub sampling to look at NTCs and low samples. Then when a cutoff was established, the data was re-run with sub sampling at 5000. 
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
#16S First pass
```{r}  
knitr::opts_chunk$set(warning=FALSE, message=FALSE)
library(plyr) 
library(dplyr)
library(ggplot2)
library(ggpubr)
library(lmerTest)
library(car)
library(emmeans)
library(gridExtra)
library(multcomp)
library(reshape)
library(factoextra)
library(reshape2)
library(vegan) 
library(pairwiseAdonis)
library("scales")
packageVersion("scales")
library(RColorBrewer)
library(colorRamps)
library(devtools)
library(phyloseq)
library(readr)
library(vegan)
library(ape)
library(geosphere)
library(ade4)
library(microbiome)  
library(knitr)
library(parzer)#parzer package to deal with coordinates
```

#Load in data and get into phyloseq (save RData file)
```{r}       
#read in sample data and merge metadata
meta_2024_canvas<-read.csv("meta/Chicago River metadata canvas 2024.csv") 
meta_2023_sampling<-read.csv("meta/Chicago River metadata 2023.csv") 
  meta_2023_sampling$sample.id<-as.character(meta_2023_sampling$sample.id)
meta_2024_sampling<-read.csv("meta/ChicagoRiver_sterivexField_Metadata_2024.csv") 
Sterivex_to_tube_2024<-read.csv("meta/SterivexCrackingMetadata_2024.csv")
qPCR_names<-read.csv("meta/Chicago River library primer plate locations.csv")

qPCR_names <- qPCR_names %>%
  mutate(project = case_when(
    Sample_Project == "Chicago River 2024" ~ "kayak",
    Sample_Project == "Chicago River 2023" ~ "zoop",
    Sample_Project == "Blank" ~ "Blank",
    Sample_Project == "APAL" ~ "APAL",
    Sample_Project == "APAL_sterivex" ~ "APAL",
    Sample_Project == "Mock" ~ "Mock",
    TRUE ~ NA_character_  # default if none match
  ))

library(dplyr)
library(stringr)

qPCR_names <- qPCR_names %>%
  mutate(sample.id_dup = if_else(
    str_detect(Sample.name, "RIV"),
    str_extract(Sample.name, "(?<=RIV)\\d+[a-zA-Z]*"),
    Sample.name
  ))

qPCR_names <- qPCR_names %>%
  mutate(sample.id_dup = case_when(
    Sample_Project == "Chicago River 2024" & str_detect(sample.id_dup, "^\\d") ~ paste0("T", sample.id_dup),
    TRUE ~ sample.id_dup
  ))

qPCR_names <- qPCR_names %>%
  mutate(sample.id = str_replace(sample.id, "b$", "")) #remove b for now, keep record

qPCR_names <- qPCR_names %>%
  left_join(Sterivex_to_tube_2024, by = "sample.id") #connect sterivex number and DNA number

qPCR_names <- qPCR_names %>%
  left_join(meta_2023_sampling, by = "sample.id")

qPCR_names <- qPCR_names %>%
  mutate(year = coalesce(year.x, year.y)) %>%
  dplyr::select(-year.x, -year.y)

qPCR_names <- qPCR_names %>%
  left_join(meta_2024_sampling, 
            by = c("sample.id" = "sample.id", 
                   "year" = "year", 
                   "project.y" = "project"))

cols_to_keep <- c( "Sample.name","Sample_Project","project.x","sample.id","sample.id_dup","date.x","sample.type","location.x","project.y","DNAtube.y","year","qPCRplate")

df <- qPCR_names %>%
  dplyr::select(all_of(cols_to_keep))




#### check if the 2024 tubes and sterivex matched up right. i think it is still off








```
Phyloseq object
```{r}
#Make phyloseq Object
sam0 <- metadata
#sam1 <- as.matrix(sam0[, -1])
sam1 <- as.matrix(sam0)
rownames(sam1) <- sam0$sample_name
sam <- sample_data(data.frame(sam1))
#write.csv(sam0, "sam.csv")


# OTU Table ####
#load in OTU: abundance_table_100.shared.txt for counts table
OTU<-read.table("SCTLD_agave_5000/16S-pipeline_outputs/Results/main/details/abundance_table_100.shared.txt", sep='', header=T)

OTU <- OTU %>%
  mutate(Group = gsub("_", ".", Group)) %>%
  mutate(Group = gsub("S\\d{3}\\.L001", "", Group)) %>%
    mutate(Group = gsub("S\\d{2}\\.L001", "", Group)) %>%
  mutate(Group = gsub("\\.$", "", Group))

otu1 <- as.matrix(OTU[, -c(1,3)]) # remove first col "label"
otu2.df<-as.data.frame(otu1) #make df copy to make samp names as row names in matrix
rownames(otu2.df) <- otu2.df$Group
otu3 <- as.matrix(otu2.df[, -(1)]) # remove first col samplename
#write.csv(otu2.df, "otu2.df.csv")
## something is preventing phyloseq from taking otu as the otu table. but if you save and then reupload it works. 
write.csv(otu3,"otucheck.csv")
testOtu<-read.csv("otucheck.csv")

#fix Sample names (typos and things not able to pull out above from meta from diff sources)
testOtu<-  testOtu %>%
  mutate(X = gsub("\\.S\\d{1,}\\.L001", "", X))

  testOtu$X[testOtu$X == "END.PS11.24"] <- "END.PS22.24"
  testOtu$X[testOtu$X == "END.PS11.37.1"] <- "END.PS22.37.1"
  testOtu$X[testOtu$X == "END.PS11.46.3"] <- "END.PS22.46.3"
  testOtu$X[testOtu$X == "END.PS11.9.S9.L001"] <- "END.PS22.9"
  testOtu$X[testOtu$X == "END.CN22.902"] <- "END.CN22.9.2"
  testOtu$X[testOtu$X == "END.PS22.10.25"] <- "END.PS22.25"
  testOtu$X[testOtu$X == "END.PS22.10.26"] <- "END.PS22.26"
    testOtu$X[testOtu$X == "END.PS22.10.26"] <- "END.PS22.26"
    testOtu$X[testOtu$X == "END.PS22.7.20"] <- "END.PS22.20"
    testOtu$X[testOtu$X == "END.PS22.7.22"] <- "END.PS22.22"
    testOtu$X[testOtu$X == "SCTLD.DNA.plate.1.A1"] <- "END.PS22.1"
    testOtu$X[testOtu$X == "SCTLD.DNA.plate.1.G1"] <- "END.CN22.21.4"
   testOtu$X[testOtu$X == "SCTLD.DNA.plate.1.H1"] <- "END.CN22.34"
   testOtu$X[testOtu$X == "SCTLD.DNA.plate.2.A1"] <- "END.DL22.1"
   testOtu$X[testOtu$X == "SCTLD.DNA.plate.2.H1"] <- "KL.OFAV.4"
   testOtu$X[testOtu$X == "SCTLD.DNA.plate.2.H2"] <- "KL.OFAV.5"
   testOtu$X[testOtu$X == "SCTLD.DNA.plate.3.A1"] <- "INI.CN22.1"
   testOtu$X[testOtu$X == "SCTLD.DNA.plate.3.A12"] <- "INI.CN22.12"

testOtu2 <- as.matrix(testOtu[, -(1)]) 
rownames(testOtu2)<- testOtu$X
otu <- otu_table(testOtu2, taxa_are_rows = FALSE)
#write.csv(otu, "checkotu.csv")

#Taxonomy Table ####
#tax table annotations_100_taxonomy.csv (edited to be in proper format with proper col names in excel (remove ";"))

#taxonomy:
TAX<- read.csv("SCTLD_agave_5000/16S-pipeline_outputs/Results/main/details/annotations_100.taxonomy.csv", colClasses = "character") 
tax1 <- as.matrix(TAX[, -1], dimnames = list(TAX$OTU, colnames(TAX[-1])))
rownames(tax1) <- TAX$OTU
tax <- tax_table(tax1)

# Read the data into phyloseq
Bac.seq = phyloseq(otu, tax,sam) #THIS WORKS
Bac.seq
Bac.seq.df <- sample_data(Bac.seq)

#load your tre file FastTree_100.nwk
treefile<- read.tree("SCTLD_agave_5000/16S-pipeline_outputs/Results/postprocessing/unifrac/FastTree_100.nwk")
phy_tree(Bac.seq) <- treefile
Bac.seq

#Save before QC if needed
save(Bac.seq, file = "RData/Bac.seq.RData")
```

Code for QC:
Done on the un-subsetted output
```{r}
# the df is Bac.seq
#note - P3/4 NCT traded!

## check out data
ntaxa(Bac.seq)  #num taxa
nsamples(Bac.seq)   #num samples
sample_names(Bac.seq) #samp names
rank_names(Bac.seq) 
sample_variables(Bac.seq) # metadata cats
sums<-sample_sums(Bac.seq)
sums<-as.data.frame(sums)
#write.csv(sums, "sums.csv")

# create df of sample data to view 
sample.data <- as(sample_data(Bac.seq), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = experiment)) +
  geom_point()

ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = experiment)) +
  geom_point()+
  facet_wrap(~experiment)

richness(Bac.seq)
plot_richness(Bac.seq, measures = c("Observed","Shannon"), color="species") 
evenness(Bac.seq)

#NTC (all NTC subsampled out due to low reads)
NTCs<-subset_samples(Bac.seq, experiment=="NTC")
TopNOTUsNTC = names(sort(taxa_sums(NTCs), TRUE)[1:200])
bac50 = prune_taxa(TopNOTUsNTC, NTCs)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot #all low

richness(NTCs)
plot_richness(NTCs, measures = c("Observed","Shannon"))
evenness(NTCs)

#Mock
Mock<-subset_samples(Bac.seq, experiment=="Mock")
TopNOTUsMock = names(sort(taxa_sums(Mock), TRUE)[1:200])
bac50 = prune_taxa(TopNOTUsMock, Mock)
bacBarPlot<-plot_bar(bac50,  fill="Phylum");bacBarPlot

richness(Mock)
plot_richness(Mock, measures = c("Observed","Shannon"))
evenness(Mock)
```

JUST SCTLD
```{r}
# look just at SCTLD samples
Bac.seq.S.2<-subset_samples (Bac.seq, experiment=="SCTLD")
#Bac.seq.S.2<-subset_samples(Bac.seq.S.2, time.point!= "bath")
Bac.seq.S.2<-subset_samples(Bac.seq.S.2, experiment!= "NTC")
Bac.seq.S.2<-subset_samples(Bac.seq.S.2, experiment!= "MOCK")
Bac.seq.S.2<-subset_samples(Bac.seq.S.2, experiment!= "FLU")
#Bac.seq.S.2<-subset_samples(Bac.seq.S.2, species!= "Orbicella faveolata")
#Bac.seq.S.2<-subset_samples(Bac.seq.S.2, species!= "Montastrea cavernosa")
Bac.seq.S.2.sd <- as(sample_data(Bac.seq.S.2), "data.frame") #look at samp data

sample.data <- as(sample_data(Bac.seq.S.2), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq.S.2)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = time.point)) +
  geom_point()+
  facet_wrap(~species)

#Check all samples same subset: Cut at 6740 seqs/samp in subset bioinformatics (<5000cut)
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = experiment)) +
  geom_point()+
  facet_wrap(~experiment)

#Save
save(Bac.seq.S.2, file = "RData/Bac.seq.SCTLD.RData")
```


JUST FLU
```{r}
# look just at SCTLD samples
Bac.seq.F.2<-subset_samples (Bac.seq, experiment=="FLU")
Bac.seq.F.2<-subset_samples(Bac.seq.F.2, time.point!= "bath")
Bac.seq.F.2<-subset_samples(Bac.seq.F.2, experiment!= "NTC")
Bac.seq.F.2<-subset_samples(Bac.seq.F.2, experiment!= "MOCK")

Bac.seq.F.2.sd <- as(sample_data(Bac.seq.F.2), "data.frame") #look at samp data

sample.data <- as(sample_data(Bac.seq.F.2), "data.frame") #create sample data frame to view
sample.data$LibrarySize <- sample_sums(Bac.seq.F.2)
sample.data <- sample.data[order(sample.data$LibrarySize),]
sample.data$Index <- seq(nrow(sample.data))  
ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = treatment)) +
  geom_point()+
  facet_wrap(~species)

ggplot(data = sample.data, aes(x=Index, y=LibrarySize, color = experiment)) +
  geom_point()+
  facet_wrap(~experiment)

save(Bac.seq.F.2, file = "/Users/smatsuda/Library/CloudStorage/OneDrive-SheddAquariumSociety/Projects/Fluconazole/Fluconazole/RData/Bac.seqFLU.RData")

```
Notes: consider rerunning 2k cutoff 

